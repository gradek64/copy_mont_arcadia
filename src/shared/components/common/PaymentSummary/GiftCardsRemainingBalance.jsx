import React from 'react'
import PropTypes from 'prop-types'
import { PrivacyGuard } from '../../../lib'
import { lastFourCharsOf } from '../../../lib/string-utils'
import { GCARD } from '../../../constants/paymentTypes'

const GiftCardsRemainingBalance = ({ payments }, { p, l }) =>
  payments
    .filter((paymentMethod) => paymentMethod.type === GCARD)
    .map((payment, i) => (
      <tr
        key={i} /* eslint-disable-line react/no-array-index-key */
        className="PaymentSummary-tableRow qa-row"
      >
        <td className="PaymentSummary-tableColumn qa-column">
          <PrivacyGuard>
            <span>
              {l`Balance left on gift card`} **** **** ****{' '}
              {lastFourCharsOf(payment.cardNumber)}
            </span>
          </PrivacyGuard>
        </td>
        <td className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column">
          <PrivacyGuard>
            <span>{p(payment.remainingBalance)}</span>
          </PrivacyGuard>
        </td>
      </tr>
    ))

GiftCardsRemainingBalance.propTypes = {
  payments: PropTypes.array.isRequired,
}

GiftCardsRemainingBalance.contextTypes = {
  l: PropTypes.func,
  p: PropTypes.func,
}

export default GiftCardsRemainingBalance
