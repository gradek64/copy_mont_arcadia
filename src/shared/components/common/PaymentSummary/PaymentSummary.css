/** @define PaymentSummary */
.PaymentSummary {
  max-width: $payment-summary-max-width;
  border-top: $payment-summary-border-top-width solid $payment-summary-border-top-color;
  border-right: $payment-summary-border-right-width solid $payment-summary-border-right-color;
  border-bottom: $payment-summary-border-bottom-width solid $payment-summary-border-bottom-color;
  border-left: $payment-summary-border-left-width solid $payment-summary-border-left-color;
  padding-top: $payment-summary-padding-top;
}

.PaymentSummary-header {
  height: 40px;
  border-bottom: 1px solid $payment-summary-header-border-bottom;
}

.PaymentSummary-headerText {
  margin: 0;
  color: $payment-summary-header-text-color;
  font-size: $payment-summary-header-font-size;
  font-weight: $payment-summary-header-font-weight;
  line-height: $payment-summary-header-line-height;
  text-transform: $payment-summary-header-text-transform;
}

.PaymentSummary-tableHead > .PaymentSummary-tableRow > .PaymentSummary-tableColumn {
  font-weight: $payment-summary-table-head-font-weight;
}

.PaymentSummary-body {
  background-color: $payment-summary-body-background-color;
  padding-top: $payment-summary-body-padding-top;
  padding-right: $payment-summary-body-padding-right;
  padding-bottom: $payment-summary-body-padding-bottom;
  padding-left: $payment-summary-body-padding-left;
}

.PaymentSummary-table {
  width: 100%;
  border-collapse: collapse;
}

.PaymentSummary-tableRow {
  padding: 0;
  font-size: $payment-summary-table-row-font-size;
  font-family: $payment-summary-table-row-font-family;
  font-weight: $payment-summary-table-row-font-weight;
  line-height: $payment-summary-table-row-line-height;
  letter-spacing: $payment-summary-table-row-letter-spacing;
  border-bottom: $payment-summary-table-row-border-bottom;
}

.PaymentSummary-tableColumn {
  padding: $payment-summary-table-col-padding;
}

.PaymentSummary-totalRow {
  text-align: left;
  & > .PaymentSummary-tableColumn {
    padding: $payment-summary-total-col-padding;
    border-bottom: $payment-summary-total-border-bottom;
  }
}


.PaymentSummary-tableColumnRight {
  text-align: right;
  white-space: nowrap;
  font-size: $payment-summary-table-col-right-font-size;
  font-weight: $payment-summary-table-col-right-font-weight;
  line-height: $payment-summary-table-col-right-line-height;
}

