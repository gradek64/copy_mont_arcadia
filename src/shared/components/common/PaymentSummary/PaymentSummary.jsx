import React from 'react'
import PropTypes from 'prop-types'
import { PrivacyGuard } from '../../../lib/privacy-guard'
import { lastFourCharsOf } from '../../../lib/string-utils'
import { GCARD } from '../../../../shared/constants/paymentTypes'
import GiftCardsRemainingBalance from './GiftCardsRemainingBalance'

export default function PaymentSummary(props, context) {
  const { totalOrderPrice, payments } = props
  const { l, p } = context
  const paymentHasGiftCards = payments.some(
    (paymentMethod) => paymentMethod.type === GCARD
  )
  return (
    <article className="PaymentSummary">
      <header className="PaymentSummary-header">
        <h1 className="PaymentSummary-headerText qa-heading">{l`Payment details`}</h1>
      </header>
      <div className="PaymentSummary-body">
        <table className="PaymentSummary-table">
          <thead className="PaymentSummary-tableHead qa-thead">
            <tr className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow">
              <th className="PaymentSummary-tableColumn qa-column">{l`Total Cost`}</th>
              <th className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column">
                {p(totalOrderPrice)}
              </th>
            </tr>
          </thead>
          <tbody className="qa-tbody">
            {payments.map((payment, i) => (
              <tr
                key={i} // eslint-disable-line react/no-array-index-key
                className="PaymentSummary-tableRow qa-row"
              >
                <td className="PaymentSummary-tableColumn qa-column">
                  <PrivacyGuard>
                    <span>{payment.method}</span>
                  </PrivacyGuard>
                  {payment.isCardType && (
                    <PrivacyGuard>
                      <span>
                        **** **** **** {lastFourCharsOf(payment.cardNumber)}
                      </span>
                    </PrivacyGuard>
                  )}
                </td>
                <td className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column">
                  <PrivacyGuard>
                    <span>
                      {p(payment.priceAfterDiscount || payment.price)}
                    </span>
                  </PrivacyGuard>
                </td>
              </tr>
            ))}
            {paymentHasGiftCards && (
              <GiftCardsRemainingBalance payments={payments} />
            )}
          </tbody>
        </table>
      </div>
    </article>
  )
}

PaymentSummary.contextTypes = {
  l: PropTypes.func,
  p: PropTypes.func,
}

PaymentSummary.propTypes = {
  payments: PropTypes.arrayOf(
    PropTypes.shape({
      method: PropTypes.string,
      cardNumber: PropTypes.string,
      price: PropTypes.string,
      type: PropTypes.string.isRequired,
    })
  ),
  totalOrderPrice: PropTypes.string,
}
