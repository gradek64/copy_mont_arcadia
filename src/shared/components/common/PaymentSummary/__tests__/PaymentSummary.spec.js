import testComponentHelper from 'test/unit/helpers/test-component'
import PaymentSummary from '../PaymentSummary'

const context = {
  l: (value) => {
    return value
  },
  p: (value) => {
    return `Â£${value}`
  },
}

const renderComponent = testComponentHelper(PaymentSummary)

describe('<PaymentSummary />', () => {
  it('should render as expected when all props have been specified', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          method: 'Visa',
          cardNumber: '************1111',
          price: '52.00',
          isCardType: true,
          type: 'VISA',
        },
        {
          method: 'Gift Card',
          cardNumber: '************1234',
          isCardType: true,
          price: '10.00',
          remainingBalance: '42.00',
          type: 'GCARD',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
  it('should render the total visa amount to be paid for order partially covered with gift card', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          method: 'Visa',
          cardNumber: '************1111',
          isCardType: true,
          price: '52.00',
          priceAfterDiscount: '32.00',
          type: 'VISA',
        },
        {
          method: 'Gift Card',
          cardNumber: '************1234',
          isCardType: true,
          price: '10.00',
          remainingBalance: '0.00',
          type: 'GCARD',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
  it('should render the total paypal amount to be paid for order partially covered with gift card', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          method: 'Paypal',
          isCardType: false,
          cardNumber: '',
          price: '52.00',
          priceAfterDiscount: '32.00',
          type: 'PYPAL',
        },
        {
          method: 'Gift Card',
          cardNumber: '************1234',
          isCardType: true,
          price: '10.00',
          remainingBalance: '0.00',
          type: 'GCARD',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
  it('should render the total klarna amount to be paid for order partially covered with gift card', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          method: 'Klarna',
          isCardType: false,
          cardNumber: '',
          price: '52.00',
          priceAfterDiscount: '32.00',
          type: 'KLRNA',
        },
        {
          method: 'Gift Card',
          cardNumber: '************1234',
          isCardType: true,
          price: '10.00',
          remainingBalance: '0.00',
          type: 'GCARD',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
  it('should render the total account card amount to be paid for order partially covered with gift card', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          cardNumber: '************0005',
          isCardType: true,
          method: 'Account Card',
          price: '8.45',
          type: 'ACCNT',
        },
        {
          method: 'Gift Card',
          cardNumber: '************1234',
          isCardType: true,
          price: '10.00',
          remainingBalance: '0.00',
          type: 'GCARD',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
  it('should only render the gift card amount paid if order total is covered by gift card', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          method: 'Gift Card',
          cardNumber: '************1234',
          isCardType: true,
          price: '52.00',
          remainingBalance: '0.00',
          type: 'GCARD',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
  it('should render the gift card remaining amount', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          method: 'Gift Card',
          cardNumber: '************1234',
          isCardType: true,
          price: '10.00',
          remainingBalance: '42.00',
          type: 'GCARD',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
  it('should not render the gift card remaining amount if no gift card is used', () => {
    const props = {
      totalOrderPrice: '52.00',
      payments: [
        {
          method: 'Visa',
          cardNumber: '************1111',
          isCardType: true,
          price: '52.00',
          type: 'VISA',
        },
      ],
    }
    const { getTree } = renderComponent(props, context)
    expect(getTree()).toMatchSnapshot()
  })
})
