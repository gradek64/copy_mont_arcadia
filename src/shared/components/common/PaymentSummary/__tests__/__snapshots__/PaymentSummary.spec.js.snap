// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentSummary /> should not render the gift card remaining amount if no gift card is used 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Visa
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1111
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £52.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</article>
`;

exports[`<PaymentSummary /> should only render the gift card amount paid if order total is covered by gift card 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Gift Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1234
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £52.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <GiftCardsRemainingBalance
          payments={
            Array [
              Object {
                "cardNumber": "************1234",
                "isCardType": true,
                "method": "Gift Card",
                "price": "52.00",
                "remainingBalance": "0.00",
                "type": "GCARD",
              },
            ]
          }
        />
      </tbody>
    </table>
  </div>
</article>
`;

exports[`<PaymentSummary /> should render as expected when all props have been specified 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Visa
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1111
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £52.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="1"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Gift Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1234
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £10.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <GiftCardsRemainingBalance
          payments={
            Array [
              Object {
                "cardNumber": "************1111",
                "isCardType": true,
                "method": "Visa",
                "price": "52.00",
                "type": "VISA",
              },
              Object {
                "cardNumber": "************1234",
                "isCardType": true,
                "method": "Gift Card",
                "price": "10.00",
                "remainingBalance": "42.00",
                "type": "GCARD",
              },
            ]
          }
        />
      </tbody>
    </table>
  </div>
</article>
`;

exports[`<PaymentSummary /> should render the gift card remaining amount 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Gift Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1234
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £10.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <GiftCardsRemainingBalance
          payments={
            Array [
              Object {
                "cardNumber": "************1234",
                "isCardType": true,
                "method": "Gift Card",
                "price": "10.00",
                "remainingBalance": "42.00",
                "type": "GCARD",
              },
            ]
          }
        />
      </tbody>
    </table>
  </div>
</article>
`;

exports[`<PaymentSummary /> should render the total account card amount to be paid for order partially covered with gift card 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Account Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                0005
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £8.45
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="1"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Gift Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1234
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £10.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <GiftCardsRemainingBalance
          payments={
            Array [
              Object {
                "cardNumber": "************0005",
                "isCardType": true,
                "method": "Account Card",
                "price": "8.45",
                "type": "ACCNT",
              },
              Object {
                "cardNumber": "************1234",
                "isCardType": true,
                "method": "Gift Card",
                "price": "10.00",
                "remainingBalance": "0.00",
                "type": "GCARD",
              },
            ]
          }
        />
      </tbody>
    </table>
  </div>
</article>
`;

exports[`<PaymentSummary /> should render the total klarna amount to be paid for order partially covered with gift card 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Klarna
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £32.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="1"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Gift Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1234
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £10.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <GiftCardsRemainingBalance
          payments={
            Array [
              Object {
                "cardNumber": "",
                "isCardType": false,
                "method": "Klarna",
                "price": "52.00",
                "priceAfterDiscount": "32.00",
                "type": "KLRNA",
              },
              Object {
                "cardNumber": "************1234",
                "isCardType": true,
                "method": "Gift Card",
                "price": "10.00",
                "remainingBalance": "0.00",
                "type": "GCARD",
              },
            ]
          }
        />
      </tbody>
    </table>
  </div>
</article>
`;

exports[`<PaymentSummary /> should render the total paypal amount to be paid for order partially covered with gift card 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Paypal
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £32.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="1"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Gift Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1234
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £10.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <GiftCardsRemainingBalance
          payments={
            Array [
              Object {
                "cardNumber": "",
                "isCardType": false,
                "method": "Paypal",
                "price": "52.00",
                "priceAfterDiscount": "32.00",
                "type": "PYPAL",
              },
              Object {
                "cardNumber": "************1234",
                "isCardType": true,
                "method": "Gift Card",
                "price": "10.00",
                "remainingBalance": "0.00",
                "type": "GCARD",
              },
            ]
          }
        />
      </tbody>
    </table>
  </div>
</article>
`;

exports[`<PaymentSummary /> should render the total visa amount to be paid for order partially covered with gift card 1`] = `
<article
  className="PaymentSummary"
>
  <header
    className="PaymentSummary-header"
  >
    <h1
      className="PaymentSummary-headerText qa-heading"
    >
      Payment details
    </h1>
  </header>
  <div
    className="PaymentSummary-body"
  >
    <table
      className="PaymentSummary-table"
    >
      <thead
        className="PaymentSummary-tableHead qa-thead"
      >
        <tr
          className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow"
        >
          <th
            className="PaymentSummary-tableColumn qa-column"
          >
            Total Cost
          </th>
          <th
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column"
          >
            £52.00
          </th>
        </tr>
      </thead>
      <tbody
        className="qa-tbody"
      >
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="0"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Visa
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1111
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £32.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <tr
          className="PaymentSummary-tableRow qa-row"
          key="1"
        >
          <td
            className="PaymentSummary-tableColumn qa-column"
          >
            <PrivacyGuard>
              <span>
                Gift Card
              </span>
            </PrivacyGuard>
            <PrivacyGuard>
              <span>
                **** **** **** 
                1234
              </span>
            </PrivacyGuard>
          </td>
          <td
            className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column"
          >
            <PrivacyGuard>
              <span>
                £10.00
              </span>
            </PrivacyGuard>
          </td>
        </tr>
        <GiftCardsRemainingBalance
          payments={
            Array [
              Object {
                "cardNumber": "************1111",
                "isCardType": true,
                "method": "Visa",
                "price": "52.00",
                "priceAfterDiscount": "32.00",
                "type": "VISA",
              },
              Object {
                "cardNumber": "************1234",
                "isCardType": true,
                "method": "Gift Card",
                "price": "10.00",
                "remainingBalance": "0.00",
                "type": "GCARD",
              },
            ]
          }
        />
      </tbody>
    </table>
  </div>
</article>
`;
